gateleen-test/src/test/java/org/swisspush/gateleen/hook/RouteListingTest.java

```diff
@@ -223,5 +224,91 @@ private void assertResponse(final Response response, final String[] expectedArra
         Assert.assertEquals(expectedArray.length, array.size());
         Assert.assertThat(array, Matchers.contains(expectedArray));
     }
+    /**
+     * Test for route listing with a valid query parameter.
+     */
+    @Test
+    public void testRouteListing_ValidQueryParam(TestContext context) {
+        Async async = context.async();
+        delete();
+        initSettings();
+
+        String queryParam = "testQuery";
+        String routePath = "/routes";
+        String requestUrl = requestUrlBase + routePath + "?q=" + queryParam;
+
+        // Register a route
+        TestUtils.registerRoute(requestUrlBase + routePath, targetUrlBase, new String[]{"GET", "POST"}, null, true, true);
+
+        // Send GET request with a valid query param
+        given().queryParam("q", queryParam)
+                .when().get(requestUrl)
+                .then().assertThat().statusCode(200);
+
+        // Validate response
+        checkGETStatusCodeWithAwait(requestUrl, 200);
```

Comment:
Check for response content

Reply:
I added an extra check for the response.